agents/helpers/graph.py

```diff
@@ -279,13 +294,15 @@ def __init__(self, agent_id):
         for x in range(-5, 6):
             for y in range(-5, 6):
                 if abs(x) + abs(y) < 6:
-                    self.nodes[self.modulate((x, y))] = Node(self.modulate((x, y)))
+                    loc = self.modulate((x, y))
+                    self.nodes[loc] = Node(self.modulate((x, y)))
 
         self.current = {agent_id: self.nodes[(0, 0)]}
         self.things = {'goals': [], 'dispensers': {}, 'taskboards': []}
         self.tasks = {}
         self.new_obs = {'obstacles': [], 'empty': [], 'agents': []}
         self.attached = []
+        self.energy = 300
```

Comment:
In graph.py we are hardcoding self.energy to 300, perhaps this could be retrieved in some other way?

Reply:
This is only because when initializing the beliefs, we do not have a percept yet. But every percept, the energy gets updated to the real amount. Therefore, if an agent would reconnect in the middle of the game having 200 energy, it would be updated in the first percept immediately.